src/dfx/src/lib/extension/manifest/extension.rs

```diff
@@ -1,21 +1,33 @@
 use serde::{Deserialize, Serialize};
 use serde_json::Value as JsonValue;
 use std::{collections::HashMap, fmt::Display};
+use url::Url;
+
+use crate::lib::error::ExtensionError;
 
 pub static MANIFEST_FILE_NAME: &str = "manifest.json";
 
+type PlatformArch = String;
+
 #[derive(Debug, Deserialize, Serialize)]
 pub struct ExtensionManifest {
-    pub name: String,
-    pub version: String,
+    pub name: Option<String>,
+    pub version: Option<String>,
```

Comment:
wrapped with `option` because in `DFX-908 Design Doc: dfx Extensions Manifests & Repo Structure`, section `External Repo Extension Database`, you can see that the name and version of the extension are moved, and instead of being entry in the collection, they are used as keys 
<img width="937" alt="image" src="https://user-images.githubusercontent.com/21069150/222502457-990d24e7-d35f-45ba-aa5c-e9b5dd41f10d.png">


the other way to solve this would be to not use `ExtensionManifest` in `ExternalExtensionManifest`, and instead use copy-paste version of `ExtensionManifest` in `src/dfx/src/lib/extension/manifest/external_extension.rs` but with the following modifications:
- `name` and `version` fields removed
- added `binaries` field (which is only relevant in the context of 3rd party registries)